main/design_documents/function-multi-versioning.md

```diff
@@ -0,0 +1,66 @@
+# Function Multi Versioning support for Arm architectures
+
+Other architectures supports [function multi versioning]
+(https://gcc.gnu.org/wiki/FunctionMultiVersioning). This documents presents
+the rationale behind the design of the same feature for Arm.
+All toolchains and platforms should use the same source code formats to simplify
+the development and deployment of software and therefore increase the use of
+architectural features. Arm architectures are a bit more complex in view of
+feature availability therefore function versions would be better to be bound to
+features instead of cpu implementations. Also the kernels may limit the
+available features in the user space. Arm architectures are a bit more complex
+in view of feature availability therefore function versions would be better to
+be bound to features instead of cpu implementations. Also the kernels may limit
+the available features in the user space. Function Multi Versioning abbreviated
+to FMV.
+
+Why ACLE instead of ABI? FMV does not change how the function behaves from the
+ABI perspective. Because the caller does not know which version is actually
+called, the following holds for all the function derived from a function via
+FMV:
+
+1. the derived function have the same signature of the original function,
+2. the derived function obey to the same calling convention of the original
+function.
+
+Today the `target` [attribute aarch64](https://gcc.gnu.org/onlinedocs/gcc/AArch64-Function-Attributes.html#AArch64-Function-Attributes>)
+is used for many purposes, some of which might overlap the functionality
+introduced by FMV. To avoid confusion, we propose the  introduction of the
+`target_version` attribute besides the `target_clones` attribute.
+
+The ACLE does not specify the underlying mechanism of the selection and the
```

Comment:
Is this true? This seems to be defined in the `Selection` section of the specs?

Reply:
The mechanics of the selection is up to the implementation but expected results are defined in the `Selection` section.